apps/desktop/desktop_native/core/src/ssh_agent/ssh_agent.rs

```diff
@@ -0,0 +1,216 @@
+// based on: https://github.com/Eugeny/russh/blob/main/russh-keys/src/agent/server.rs
+
+use std::collections::HashMap;
+use std::marker::Sync;
+use std::sync::{Arc, RwLock};
+
+use byteorder::{BigEndian, ByteOrder};
+use futures::stream::{Stream, StreamExt};
+use russh_cryptovec::CryptoVec;
+use ssh_key::{HashAlg, SigningKey};
+use crate::ssh_agent::russh_encoding::{Encoding, Reader, Position};
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
+
+use crate::ssh_agent::msg;
+use anyhow::Error;
+
+use super::msg::{REQUEST_IDENTITIES, SIGN_REQUEST};
+use std::result::Result;
+
+#[derive(Clone)]
+pub struct Key {
+    pub private_key: Option<ssh_key::private::PrivateKey>,
+    pub name: String,
+    pub cipher_uuid: String,
+}
+
+#[derive(Clone)]
+#[allow(clippy::type_complexity)]
+pub struct KeyStore(pub Arc<RwLock<HashMap<Vec<u8>, Key>>>);
+
+#[allow(missing_docs)]
+#[derive(Debug)]
+pub enum ServerError<E> {
+    E(E),
+    Error(Error),
+}
+
+pub trait Agent: Clone + Send + 'static {
+    fn confirm(&self, _pk: Key) -> impl std::future::Future<Output = bool> + std::marker::Send {
```

Comment:
Tiny nit: Send should be in the rust prelude and so it doesn't need to be fully qualified.
```suggestion
    fn confirm(&self, _pk: Key) -> impl std::future::Future<Output = bool> + Send {
```